### PR TITLE
Adding assignment 5

### PR DESCRIPTION
**Pull request by James-S-Santangelo**

### CODE REVIEW

File: assignment-05.Rmd [hunk 1]
```diff
@@ -1,145 +1,269 @@
...
+were fully crossed for a total of 8 treatment combinations. In each treatment combination, they grew 
+4 individuals from each of 25 plant genotypes for a total of 800 plants (8 treatment combinations x 
+25 genotypes x 4 individuals per genotype). Plants were grown in a common garden at the Koffler 
+Scientific Reserve (UofTs field research station) and 6 floral traits were measured on all plants 
+throughout the summer. We will analyze how the treatments influenced one of these traits in this 
+exercise. Run the code chunk below to download the data, which includes only a subset of the columns 
+from the full dataset:
+    
+```{r}
+plant_data <- read.csv("data/Santangelo_JEB_2018.csv", header = TRUE)
```
Comment (aays) [inline assignment-05.Rmd]: Can we use `readr::read_csv` here? Just to be consistent with previous lectures/assignments

File: assignment-05.Rmd [hunk 2]
```diff
@@ -1,145 +1,269 @@
...
+model <- lmer(Total.Inf ~ HCN*Herbivory + (1|Genotype) + (1|Block), data = plant_data)
+```
+
+(b) Summarize (i.e. get the output) the model that you ran in part (a). Did any of the treatments have a significant effect on the number of inflorescences? If so, which one? Based on your examination of the model output, how can you tell which level of the significant treatment resulted in higher of lower mean inflorescnece production? (1 mark).
+    
+```{r include=FALSE}
+summary(model)
+```
+
+(c) Using `dplyr` and `gglot2`, plot the mean inflorescence production plus or minus one standard error against the treatment that showed a significant effect in the model above. Feel free to use whatever kind of plot you would that is appropriate to this king of data [**Hint:** you should summarize the data in `dplyr` prior to plotting with `ggplot()`]. As a reminder, I have inluded the formula to calculate the standard error of the mean below. (1 mark).
```
Comment (aays) [inline assignment-05.Rmd]: *kind of data?

File: assignment-05.Rmd [hunk 3]
```diff
@@ -1,145 +1,269 @@
...
+model <- lm(y ~ x1 + x2)
+summary(model)
+```
+
+**3. Randomization test (2 marks)**
+
+Run the code chunk below to load the data that will be used in this exercise.
+
+```{r}
+df <- read.csv("data/Assign05_Question3.csv")
```
Comment (aays) [inline assignment-05.Rmd]: Same as above re: `readr::read_csv`

File: assignment-05.Rmd [hunk 4]
```diff
@@ -1,145 +1,269 @@
...
+```{r include = FALSE}
+abs_simulated_medians <- abs(simulated_medians)
+abs_diff_medians_obs <- abs(diff_median_obs)
+exceed_count <- length(abs_simulated_medians[abs_simulated_medians >= 
+                                               abs_diff_medians_obs])
+p_val <- exceed_count / nreps
+p_val
+```
+
+**4. Bonus: Bootsrapping (2 marks)** In lecture, I mentioned that randomization tests are frequently used to estimate the amount of variance surrounding a population parameter. This is often done using **bootstrapping**, in which samples are repeatedly sampled **with replacement** from the observed data to estimate the distribution of the population statitic, from which the variance (or other measure of spread) can be estimated. In this _bonus_ exercise, I ask you to estimate the 95% confidence intervals around the _beta coefficient_ for the effect of variable `x2` on `y` in the simulated regression from question 2.
```
Comment (aays) [inline assignment-05.Rmd]: *population statistic

File: assignment-05.Rmd [hunk 5]
```diff
@@ -1,145 +1,269 @@
...
+
+(a) Generate a histogram showing the distribution of beta coefficients estimated by re-running the linear model in question 2 with different values for variable `x2` obtained by resampling the original `x2` distribution with replacement. You should perform 5,000 simulations of the linear model, each time saving the beta coefficient for the effect of the variable `x2` on `y`. The `y_intercept`, variable `x1` and the `error` should remain unchanged during each iteration. A single iteration should look like the following: (1) Resample 20 values of variable `x2` with replacement. (2) Run linear model with `y` as a linear function of `x1` and the newly resampled `x2` as in question 2. (3) Extract beta coefficient for effect of `x2` on `y` from the linear model and add to list. Repeat 5,000 times. Plot the distribution of betas. [**hint:** You can get the coefficients from a linear model using `coef(summary(model))`. You'll have to figure out how to extract the relavent beta using this command] (1 mark)
```
Comment (aays) [inline assignment-05.Rmd]: *relevant beta

Comment (James-S-Santangelo) [reply assignment-05.Rmd]: I always spell relevant wrong...

File: assignment-05.Rmd [hunk 6]
```diff
@@ -1,145 +1,269 @@
...
-    -   What is the biological meaning of $0 < \alpha < 1$? What is the
-        biological meaning of $\alpha = 1$, and of $\alpha > 1$? (Hint:
-        what is the sum of the non-trivial fixed points $x^*$ and $y^*$
-        in each of these cases in terms of $K_1$ and $K_2$?)
-
-    -   What would happen to the system if $\alpha$ was negative? (Hint:
-        is "competition" still an appropriate name for the model if
-        $\alpha < 0$?)
+including the original questions, your code, and the output. Submit
+you assignment as a knitted `.pdf` (prefered) or `.html` file.*
```
Comment (QuLogic) [inline assignment-05.Rmd]: s/prefered/preferred/

File: assignment-05.Rmd [hunk 7]
```diff
@@ -1,145 +1,269 @@
...
+plant defenses (`HCN`) _and their interaction_ influences the total number of inflorescences (`Total.Inf`) produced by
+plants while accounting for variation due to spatial block and plant genotype (1 mark: 0.5 for correct fixed effects specification and 0.5 for correct random effects structure). You only need to specify the model for this part of the question.
+
+```{r include=FALSE}
+library(lme4)
+library(lmerTest)
+model <- lmer(Total.Inf ~ HCN*Herbivory + (1|Genotype) + (1|Block), data = plant_data)
+```
+
+(b) Summarize (i.e. get the output) the model that you ran in part (a). Did any of the treatments have a significant effect on the number of inflorescences? If so, which one? Based on your examination of the model output, how can you tell which level of the significant treatment resulted in higher of lower mean inflorescnece production? (1 mark).
```
Comment (QuLogic) [inline assignment-05.Rmd]: s/inflorescnece/inflorescence/

File: assignment-05.Rmd [hunk 8]
```diff
@@ -1,145 +1,269 @@
...
+    
+```{r include=FALSE}
+summary(model)
+```
+
+(c) Using `dplyr` and `gglot2`, plot the mean inflorescence production plus or minus one standard error against the treatment that showed a significant effect in the model above. Feel free to use whatever kind of plot you would that is appropriate to this kind of data [**Hint:** you should summarize the data in `dplyr` prior to plotting with `ggplot()`]. As a reminder, I have inluded the formula to calculate the standard error of the mean below. (1 mark).
```
Comment (QuLogic) [inline assignment-05.Rmd]: s/inluded/included.

File: assignment-05.Rmd [hunk 9]
```diff
@@ -1,145 +1,269 @@
...
+
+shape <- 1.25
+rate <- 3.5
+n <- 100
+x <- rgamma(n, shape, rate)
+
+hist(x)
+```
+
+(b) Plot the distribution of sample means obtained by generating 1000 gamma distributions with the same parameters as in (a). In other words, the distribution should be made up of 1000 means, each from a different simulated gamma distribution. What do you notice about this distribution when conpared to the original distribution in (a)? Why would we expect this? (0.5 marks)
```
Comment (QuLogic) [inline assignment-05.Rmd]: s/conpared/compared/

**Review by QuLogic — COMMENTED** [review]
Some typos.

File: assignment-05.Rmd (hunk 6 above)
Comment (aays) [reply assignment-05.Rmd]: Also *your assignment

File: assignment-05.Rmd [hunk 10]
```diff
@@ -1,145 +1,269 @@
...
+were fully crossed for a total of 8 treatment combinations. In each treatment combination, they grew 
+4 individuals from each of 25 plant genotypes for a total of 800 plants (8 treatment combinations x 
+25 genotypes x 4 individuals per genotype). Plants were grown in a common garden at the Koffler 
+Scientific Reserve (UofTs field research station) and 6 floral traits were measured on all plants 
+throughout the summer. We will analyze how the treatments influenced one of these traits in this 
+exercise. Run the code chunk below to download the data, which includes only a subset of the columns 
+from the full dataset:
+    
+```{r}
+plant_data <- readr::read_csv("data/Santangelo_JEB_2018.csv", header = TRUE)
```
Comment (aays) [inline assignment-05.Rmd]: Seems `header = TRUE` is breaking Travis - I think the `readr` equivalent is `col_names`, but it's also set to `TRUE` by default, so no second argument is needed

**Comment by James-S-Santangelo** [comment]
Will fix all of this tonight.

File: assignment-05.Rmd [hunk 11]
```diff
@@ -23,10 +23,11 @@ throughout the summer. We will analyze how the treatments influenced one of thes
 exercise. Run the code chunk below to download the data, which includes only a subset of the columns 
 from the full dataset:
     
-```{r}
-plant_data <- readr::read_csv("data/Santangelo_JEB_2018.csv", 
+```{r message=FALSE, warning=FALSE}
+library(tidyverse)
+plant_data <- read_csv("data/Santangelo_JEB_2018.csv", 
                               col_names = TRUE)
```
Comment (QuLogic) [inline assignment-05.Rmd]: Re-align?

File: lec08-linear-mixed-effects-models.Rmd [hunk 12]
```diff
@@ -59,19 +59,21 @@ A diagrammatic representation of the dataset can be seen below (Modified from Zu
...
 rikz_data <- paste0("https://raw.githubusercontent.com/James-S-Santangelo/",
                     "rcourse/master/data/rikz_data.txt")
 download.file(rikz_data, "rikz_data.txt")
 
-rikz_data <- read.table("rikz_data.txt",
-                        head = TRUE,
-                        sep = "\t")
+rikz_data <- read_delim("rikz_data.txt",
+                         col_names = TRUE,
+                         delim = "\t")
```
Comment (QuLogic) [inline lec08-linear-mixed-effects-models.Rmd:72]: One space too many on the wrapped lines.

**Pull request merged**